pkg/volume/util/fsquota/common/quota_linux_common_impl.go

```diff
@@ -126,8 +126,8 @@ func doRunXFSQuotaCommand(mountpoint string, mountsFile, command string) (string
 	}
 	// We're using numeric project IDs directly; no need to scan
 	// /etc/projects or /etc/projid
-	klog.V(4).Infof("runXFSQuotaCommand %s -t %s -P/dev/null -D/dev/null -x -f %s -c %s", quotaCmd, mountsFile, mountpoint, command)
-	cmd := exec.Command(quotaCmd, "-t", mountsFile, "-P/dev/null", "-D/dev/null", "-x", "-f", mountpoint, "-c", command)
+	klog.V(4).Infof("runXFSQuotaCommand %s -t %s -P/dev/null -D/dev/null -x %s -c %s", quotaCmd, mountsFile, mountpoint, command)
+	cmd := exec.Command(quotaCmd, "-t", mountsFile, "-P/dev/null", "-D/dev/null", "-x", mountpoint, "-c", command)
```

Comment:
`-f` is needed on ext4 filesystems with project quotas enabled, which is a valid case.  It isn't needed on actual xfs filesystems.

Reply:
```
switch fsTypeMagic {
	case unix.EXT4_SUPER_MAGIC:
		// We're using numeric project IDs directly; no need to scan
		// /etc/projects or /etc/projid
		klog.V(4).Infof("runXFSQuotaCommand %s -t %s -P/dev/null -D/dev/null -x -f %s -c %s", quotaCmd, mountsFile, mountpoint, command)
		cmd = exec.Command(quotaCmd, "-t", mountsFile, "-P/dev/null", "-D/dev/null", "-x", "-f", mountpoint, "-c", command)
		break
	case unix.XFS_SUPER_MAGIC:
		klog.V(4).Infof("runXFSQuotaCommand %s -t %s -P/dev/null -D/dev/null -x %s -c %s", quotaCmd, mountsFile, mountpoint, command)
		cmd = exec.Command(quotaCmd, "-t", mountsFile, "-P/dev/null", "-D/dev/null", "-x", mountpoint, "-c", command)
		break
	default:
		return "", fmt.Errorf("unsupported fs type: %x", fsTypeMagic)
```

Reply:
The comment about project IDs belongs above the switch.
That approach is reasonable; just note that `-f` is needed for any non-XFS filesystem.  If someone ports XFS quotas to another filesystem (e. g. btrfs), you probably don't want to add one more bit of code that has to be changed.  So you might simply want to make that be the default case.

Reply:
```go
// We're using numeric project IDs directly; no need to scan
// /etc/projects or /etc/projid
switch fsTypeMagic {
// xfs don't need '-f' option
case unix.XFS_SUPER_MAGIC:
    klog.V(4).Infof("runXFSQuotaCommand %s -t %s -P/dev/null -D/dev/null -x %s -c %s", quotaCmd, mountsFile, mountpoint, command)
    cmd = exec.Command(quotaCmd, "-t", mountsFile, "-P/dev/null", "-D/dev/null", "-x", mountpoint, "-c", command)
    break
// other filesystems (such as ext4) need '-f' option
default:
    klog.V(4).Infof("runXFSQuotaCommand %s -t %s -P/dev/null -D/dev/null -x -f %s -c %s", quotaCmd, mountsFile, mountpoint, command)
    cmd = exec.Command(quotaCmd, "-t", mountsFile, "-P/dev/null", "-D/dev/null", "-x", "-f", mountpoint, "-c", command)
}
```